src/Core/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -57,27 +58,24 @@ public class OpenApiDocumentor : IOpenApiDocumentor
         public const string DOCUMENT_CREATION_UNSUPPORTED_ERROR = "OpenAPI description document can't be created when the REST endpoint is disabled globally.";
         public const string DOCUMENT_CREATION_FAILED_ERROR = "OpenAPI description document creation failed";
 
-        public void DocumentorOnConfigChanged(object? sender, HotReloadEventArgs args)
-        {
-            Console.BackgroundColor = ConsoleColor.Yellow;
-            Console.WriteLine($"[OpenApiDocumentor]: Received event with message: {args.Message}");
-            Console.ResetColor();
-            CreateDocument(doOverrideExistingDocument: true);
-        }
-
         /// <summary>
         /// Constructor denotes required services whose metadata is used to generate the OpenAPI description document.
         /// </summary>
         /// <param name="sqlMetadataProvider">Provides database object metadata.</param>
         /// <param name="runtimeConfigProvider">Provides entity/REST path metadata.</param>
         public OpenApiDocumentor(IMetadataProviderFactory metadataProviderFactory, RuntimeConfigProvider runtimeConfigProvider, HotReloadEventHandler<HotReloadEventArgs> handler)
         {
-            handler.DocumentorSubscribe(DocumentorOnConfigChanged);
+            handler.Subscribe(DOCUMENTOR_ON_CONFIG_CHANGED, OnConfigChanged);
```

Comment:
Looks like for OpenApiDocumentor constructor, will we always need and have this handler? So, is no need to make it nullable like the others?

Reply:
As of now we will always have the handler when we make a new documentor, but it might be the case that we want to construct this class without a handler, in particular if we add tests that want to create a documentor, it would be nice if we don't necessarily have to mock this, so I will make it nullable for convenience.